src/scene/shader-lib/chunk-builder.js

```diff
@@ -1,24 +1,46 @@
+/**
+ * Helper function to ensure a bit of backwards compatibility
+ * @example
+ * toLitArgs('litShaderArgs.sheen.specularity'); // Result: 'litArgs_sheen_specularity'
+ * @param {string} src - The shader source which may generate shader errors.
+ * @returns {string} The backwards compatible shader source.
+ */
+const toLitArgs = src => src.replace(/litShaderArgs([\.a-zA-Z]+)+/g, (a, b) => {
+    return 'litArgs' + b.replace(/\./g, '_');
+});
+
 // helper class for combining shader chunks together
 // ensures every chunk ends with a new line otherwise shaders can be ill-formed
 class ChunkBuilder {
-    code = '';
+    _code = '';
+
+    set code(newCode) {
+        this._code = newCode;
+    }
+
+    get code() {
+        if (this._code.includes('litShaderArgs')) {
+            this._code = toLitArgs(this._code);
```

Comment:
Personally I'm keen to add it. We discussed how to auto upgrade shaders in the past, but in most cases, the solution would not be simple.
But in this case, I don't see why not.

I'd add `Debug.deprecated` there to inform the user to update the chunks, instead of just silently fixing this forever.

any thoughts @slimbuck ?

Reply:
```js
/*
 * @see https://github.com/playcanvas/engine/pull/5463
 */
const toLitArgs = src => src.replace(/litShaderArgs([\.a-zA-Z]+)+/g, (a, b) => {
    const newSource = 'litArgs' + b.replace(/\./g, '_');
    console.warn("Nested struct property access is deprecated, because it's crashing some devices. Please update your custom chunks manually. In particular", a, "should be", newSource);
    return newSource;
});

toLitArgs('litShaderArgs.sheen.specularity'); // Result: 'litArgs_sheen_specularity'
```

Maybe overkill? But I guess as informative as it gets:

![image](https://github.com/playcanvas/engine/assets/5236548/5ab25cf8-b3b3-4619-8f3c-699f805a64d3)

Other suggestions?

Reply:
Only thing I would raise is performance. How expensive is it to run these kinds of regexes, particularly on a large fragment shader?

Reply:
I was thinking about this, and would suggest a change. I'd move all this backwards compatibility for this class to deprecated.js - perhaps to a function `ChunkBuilder.handleCompatibility`, and call it from here if it exists. That would allow us to have it excluded for module / tree-shaking builds, where you manually update to a new minor version, and so have a control over when you fix the chunks. And we would not have this function as part of the main engine in that case too.

Reply:
@mvaligursky Thank you for the feedback, I wasn't too happy about using `ChunkBuilder` either, it was just the first solution to be able to continue with other problems and I realized that the `ChunkBuilder#code` was accessed way more often than I thought (causing a bunch of needless `String#includes` checks). Now I just patch `LitShader#+generateFragmentShader` in case `deprecated.js` is included. What do you think about this?